### ISSUE TITLE
Check that dead-but-not-removed agents are being handled sensibly

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#296]
if we implement this change, wouldn't we need to go through all the modules and make sure that we were always conditioning on people being alive before selecting them for e.g. pair formation, disease acquisition/progression, etc?

_Originally posted by @robynstuart in https://github.com/amath-idm/starsim/issues/294#issuecomment-1958601482_

**Comment by cliffckerr** [comment]
There are dead agents up to step 10:
```py
import starsim as ss

def check_dead(sim):
    print(sum(~sim.people.alive))
    
sim = ss.demo(analyzers=check_dead).run()
```

**Comment by cliffckerr** [comment]
This **is** a bug: add these lines to the end of `Infection.make_new_cases()`:
```py
                assert sum(~sim.people.alive[trg]) == 0
                assert sum(~sim.people.alive[src]) == 0
```
Then rerun. Dead people can be sources of infection.

**Comment by cliffckerr** [comment]
Closed by #456
